### PR TITLE
Fix multi waypoint leg bug

### PR DESCRIPTION
**Pull request by bsudekum**
Closes: https://github.com/mapbox/mapbox-navigation-ios/issues/798

Fixes an issue where legs would not show up.

/cc @1ec5

### CODE REVIEW

**Comment by bsudekum** [comment]
simulator screen shot - iphone x - 2017-11-07 at 17 51 28

eh, a little off.

File: MapboxNavigation/NavigationMapView.swift [hunk 1]
```diff
@@ -577,12 +576,19 @@ open class NavigationMapView: MGLMapView {
...
+                $0.coordinates?.suffix(from: 1)
+            }.joined())
+            
+            // We need to add the first coord of the route in.
+            if let firstCoord = leg.steps.first?.coordinates?.first {
+                legCoordinates.insert(firstCoord, at: 0)
+            }
             
-            previousLegCongestionIndex = legCongestion.count
+            let segments = zip(legCoordinates, legCoordinates).map { [$0.0, $0.1] }
```
Comment (1ec5) [inline MapboxNavigation/NavigationMapView.swift]: Zipping `legCoordinates` with itself is pointless. I think this is the source of the off-by-one errors you’re seeing.

Comment (1ec5) [reply MapboxNavigation/NavigationMapView.swift]: Also, a tuple is easy enough to work with that I don’t think it’s necessary to map it to an array containing the same values.

**Comment by bsudekum** [comment]
Fixed: 

simulator screen shot - iphone x - 2017-11-08 at 09 54 21

File: MapboxNavigation/NavigationMapView.swift [hunk 2]
```diff
@@ -577,12 +576,19 @@ open class NavigationMapView: MGLMapView {
...
+                $0.coordinates?.suffix(from: 1)
+            }.joined())
+            
+            // We need to add the first coord of the route in.
+            if let firstCoord = leg.steps.first?.coordinates?.first {
+                legCoordinates.insert(firstCoord, at: 0)
+            }
             
-            previousLegCongestionIndex = legCongestion.count
+            let segments = zip(legCoordinates, legCoordinates.suffix(from: 1)).map { [$0.0, $0.1] }
```
Comment (1ec5) [inline MapboxNavigation/NavigationMapView.swift]: Remove the call to `map(_:)`. That will obviate at least one `if let` below.

File: MapboxNavigation/NavigationMapView.swift [hunk 3]
```diff
@@ -577,12 +576,19 @@ open class NavigationMapView: MGLMapView {
...
+            let congestionSegments = Array(zip(segments, legCongestion))
```
Comment (1ec5) [inline MapboxNavigation/NavigationMapView.swift]: Can I bother you to add https://github.com/mapbox/mapbox-navigation-ios/pull/807#pullrequestreview-74974713 as comments?

**Pull request merged**